x-pack/solutions/observability/plugins/observability_agent/server/attachments/ai_insight.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import dedent from 'dedent';
+import { safeJsonStringify } from '@kbn/std';
+import type { Attachment } from '@kbn/onechat-common/attachments';
+import type { AttachmentTypeDefinition } from '@kbn/onechat-server/attachments';
+import {
+  OBSERVABILITY_GET_SERVICES_TOOL_ID,
+  OBSERVABILITY_GET_DOWNSTREAM_DEPENDENCIES_TOOL_ID,
+} from '../../common/constants';
+import { OBSERVABILITY_GET_ALERTS_TOOL_ID } from '../tools/get_alerts/get_alerts';
+
+export const OBSERVABILITY_AI_INSIGHT_ATTACHMENT_TYPE_ID = 'observability.ai_insight';
+
+const aiInsightAttachmentDataSchema = z.object({
+  contextData: z.record(z.any()),
+  aiSummary: z.string(),
```

Comment:
No strong feelings but perhaps less is more

```suggestion
  context: z.record(z.any()),
  summary: z.string(),
```